lib/ui/painting/picture.cc

```diff
@@ -111,17 +171,16 @@ Dart_Handle Picture::RasterizeToImage(sk_sp<SkPicture> picture,
     delete image_callback;
   });
 
-  // Kick things off on the GPU.
-  fml::TaskRunner::RunNowOrPostTask(
-      gpu_task_runner,
-      [ui_task_runner, snapshot_delegate, picture, picture_bounds, ui_task] {
-        sk_sp<SkImage> raster_image =
-            snapshot_delegate->MakeRasterSnapshot(picture, picture_bounds);
-
-        fml::TaskRunner::RunNowOrPostTask(
-            ui_task_runner,
-            [ui_task, raster_image]() { ui_task(raster_image); });
-      });
+  fml::TaskRunner::RunNowOrPostTask(io_task_runner, [ui_task_runner, picture,
+                                                     picture_bounds, ui_task,
+                                                     resource_context] {
+    sk_sp<SkSurface> surface =
+        MakeSnapshotSurface(picture_bounds, resource_context);
+    sk_sp<SkImage> raster_image = MakeRasterSnapshot(picture, surface);
+
+    fml::TaskRunner::RunNowOrPostTask(
```

Comment:
`RunNowOrPostTask` is only used if there is some form of latching involved. There isn't any here. Please post the task directly to the UI task runner. I realize this may have been because the previous code did the same. But there really is no reason to do this here.

Reply:
Done.

Reply:
I had to revert this too.  It was causing a deadlock for some tests.